BugFix: Handling amount entered when creating a new expense

trentfitzgibbon opened the pull request:
## Context
During the SAP Cloud Platform SDK for iOS Workshop I noticed a bug in the sample code when entering new expenses. None of my new expenses had values after submitting.

## Issue
The problem is related to how changed values are passed to FUITitleFormCell in the onChangeHandler if formatting is used for the valueTextField text value for display. Depending on the timing of cell reload and whether the user uses the enter key or tab to move out of field, the change handler may get the raw or formatted version of the new value. The previous code assumed no formatting would ever get passed for the change value.

## Fix
The fix handles 3 possible scenarios:
- A formatted value using the same formatter as used for display
- A simple double value string (previous code case)
- No value, the user has cleared the field after previously setting value

trentfitzgibbon committed:
Fix for handling amount value when creating an expense

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/SAP/fiori-for-ios-workshop?pullRequest=5) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/SAP/fiori-for-ios-workshop?pullRequest=5) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/SAP/fiori-for-ios-workshop?pullRequest=5) it.</sub>

sstadelman commented:
Trent, thanks for the patch. I can review and hopefully merge on Monday.

trentfitzgibbon committed:
Removed double conversion to keep precision

trentfitzgibbon commented:
@sstadelman I realised after creating this PR that the extra double value i added was unnecessary and potentially could result in floating point precision loss. So I reverted to getting BigDecimal to do the string parse. I think initially i just liked the consistent BigDecimal construction on rhs of all 3 cases 😂
